src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,125 @@
+            resize();
+        }
+        table = putElement(key, value, table);
     }
 
     @Override
     public V getValue(K key) {
+        Node<K, V> curentNode = table[getIndex(key)];
+        while (curentNode != null) {
+            if (campareKeys(curentNode.key, key)) {
+                return curentNode.value;
+            }
+            curentNode = curentNode.next;
+        }
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private void resize() {
+        Node<K, V>[] oldTable = table;
+        table = new Node[table.length << 1];
+        size = 0;
+        for (int i = 0; i < oldTable.length; i++) {
+            Node<K, V> curentNode = oldTable[i];
+            while (curentNode != null) {
+                table = putElement(curentNode.key, curentNode.value, table);
+                curentNode = curentNode.next;
+            }
+        }
+    }
+
+    private Node<K, V>[] putElement(K key, V value, Node<K, V>[] table) {
+        Node<K, V> newElement = new Node<>(key, value);
+        int index = getIndex(key);
+        if (table[index] != null) {
+            if (isKeyPresent(key, table)) {
+                changeElementInNodeList(table[index], newElement);
+                return table;
+            }
+            putToNodeList(table[index], newElement);
+            size++;
+            return table;
+        }
+        table[index] = newElement;
+        size++;
+        return table;
+    }
+
+    private void putToNodeList(Node<K, V> curentNode, Node<K, V> addNode) {
+        Node<K, V> node = curentNode;
+        while (node.next != null) {
+            node = node.next;
+        }
+        node.next = addNode;
+    }
+
+    private void changeElementInNodeList(Node<K, V> curentNode, Node<K, V> addNode) {
+        Node<K, V> node = curentNode;
+        while (node != null) {
+            if (campareKeys(node.key, addNode.key)) {
+                node.value = addNode.value;
+                return;
+            }
+            node = node.next;
+        }
+    }
+
+    private int getIndex(K key) {
+        return key == null ? 0 : Math.abs(key.hashCode() % table.length);
+    }
+
+    private boolean isKeyPresent(K key, Node<K, V>[] table) {
+        for (int i = 0; i < table.length; i++) {
+            if (table[i] != null) {
+                Node<K, V> curentNode = table[i];
+                while (curentNode != null) {
+                    if (campareKeys(curentNode.key, key)) {
+                        return true;
+                    }
+                    curentNode = curentNode.next;
+                }
+            }
+        }
+        return false;
+    }
+
+    private boolean campareKeys(K firstKey, K secondKey) {
+        return (firstKey == secondKey) || (firstKey != null) && firstKey.equals(secondKey);
+    }
+
+    private boolean isMapLoad() {
+        return size == (double)table.length * LOAD_FACTOR;
```

Comment:
Use > or < instead of ==
you don't need to cast, cause it's hard to check double values on equals. Don't forget that `0.1 + 0.2 != 0.3`.`table.length * LOAD_FACTOR` is enough